### PR TITLE
Fix enclosing variable redefinition

### LINKED ISSUE #5576: reimport with files is broken
No unit test coverage for reimport it seems.

image

### PR DESCRIPTION
**Pull request by damiencarol**
# Work done

- [x] Fix nested 'for' statement uses loop variable 'item' of enclosing for statement (line 54).

https://github.com/DefectDojo/django-DefectDojo/blob/ba1a415219ff20e8b4e909ef14f750de9b80297e/dojo/tools/generic/parser.py#L32-L64

- [x] add unit test to cover #5576 issue
- [x] fixes #5576

### CODE REVIEW

File: unittests/test_import_reimport.py [hunk 1]
```diff
@@ -1143,6 +1145,35 @@ def test_import_param_close_old_findings_with_additional_endpoint(self):
         engagement_findings_count = Finding.objects.filter(test__engagement_id=1, test__test_type=test.test_type, active=True, is_mitigated=False).count()
         self.assertEqual(engagement_findings_count, 0)
 
+    def test_import_reimport_generic(self):
```
Comment (StefanFl) [inline unittests/test_import_reimport.py:1148]: Should this test check, if the files are processed properly or is this not necessary?

Comment (damiencarol) [reply unittests/test_import_reimport.py:1148]: This test use a part of the code that were not covered before (having FileUpload in a finding).
This part fail without the patch of this PR

Comment (damiencarol) [reply unittests/test_import_reimport.py:1148]: This unit test failing shows that: https://github.com/DefectDojo/django-DefectDojo/pull/5632/commits/ed13a02c3ab289b8444dfb04e5359b55b9fe5119

Comment (StefanFl) [reply unittests/test_import_reimport.py:1148]: Great, just wanted to make sure that files are covered.

**Review by StefanFl — APPROVED** [review]

**Review by valentijnscholten — APPROVED** [review]

**Pull request merged**